### PR TITLE
[classifier] Provide a more efficient classification mechanism

### PR DESCRIPTION
**Pull request by akyrtzi**
Provide 'classifications' as a sequence of (kind, range) values.
This allows using the same type to classify source ranges across tokens and trivia pieces and makes the classification mechanism much more efficient since we avoid providing syntax nodes and the resulting casts to `_SyntaxBase` for establishing the parent hierarchy.

The new classification mechanism is 10.2x faster than the previous mechanism, and 2.7x faster than using the SyntaxVisitor.

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: Sources/SwiftSyntax/Syntax.swift [hunk 1]
```diff
@@ -464,6 +479,20 @@ extension Syntax {
...
+  public var classifications: SyntaxClassifications {
+    return base.classifications
+  }
+
+  /// Sequence of `SyntaxClassifiedRange`s contained in this syntax node within
+  /// a relative range.
+  /// - Parameters:
+  ///   - in: The relative byte range to pull `SyntaxClassifiedRange`s from.
+  /// - Returns: Sequence of `SyntaxClassifiedRange`s.
+  public func classifications(in range: ByteSourceRange) -> SyntaxClassifications {
```
Comment (nkcsgexi) [inline Sources/SwiftSyntax/Syntax.swift]: Could we migrate the location structures here to `SourceRange`?

Comment (akyrtzi) [reply Sources/SwiftSyntax/Syntax.swift]: It's easy to get a `ByteSourceRange` from `SourceRange`, the other way is not so convenient.
I would prefer not to require a `SourceRange` in this API but just add a `var byteRange: ByteSourceRange` property to `SourceRange`. If you have a `SourceRange` just pass its `byteRange` property to this API.

**Pull request merged**